File: data/src/main/java/com/rubik/shifttest/data/data/storage/sharedprefs/SharedPrefUserStorage.java
package com.rubik.shifttest.data.data.storage.sharedprefs;

import android.content.Context;
import android.content.SharedPreferences;

import com.google.gson.Gson;

import com.rubik.shifttest.data.data.storage.UserStorage;
import com.rubik.shifttest.data.data.storage.models.User;

public class SharedPrefUserStorage implements UserStorage {


    public static final String SHARED_PREFS_USER_REGISTER_CREDENTIAL = "user_register_credential";
    public static final String KEY_USER_REGISTER_CREDENTIAL = "key_user_register_credential";
    private final Context context;
    private final SharedPreferences sharedPreferences;

    public SharedPrefUserStorage(Context context) {
        this.context = context;
        sharedPreferences = this.context.getSharedPreferences(SHARED_PREFS_USER_REGISTER_CREDENTIAL, android.content.Context.MODE_PRIVATE);
    }
    @Override
    public boolean saveUser(User user) {
        Gson gson = new Gson();
        sharedPreferences.edit().putString(KEY_USER_REGISTER_CREDENTIAL, gson.toJson(user)).apply();
        return true;
    }

    @Override
    public User getUser() {
        Gson gson = new Gson();
        String json = sharedPreferences.getString(KEY_USER_REGISTER_CREDENTIAL, null);
        return (json == null) ? null : gson.fromJson(json, User.class);
    }
}

File: shared/user/src/main/kotlin/ru/rubik/user/data/storage/sharedprefs/SharedPrefUserStorage.kt
package ru.rubik.user.data.storage.sharedprefs

import android.content.Context
import android.content.SharedPreferences
import com.google.gson.Gson
import ru.rubik.user.data.storage.UserStorage
import ru.rubik.user.data.storage.models.UserDTO

class SharedPrefUserStorage(context: Context) : UserStorage {

	private val sharedPreferences: SharedPreferences = context.getSharedPreferences(SHARED_PREFS_USER_REGISTER_CREDENTIAL, Context.MODE_PRIVATE)

	override fun saveUser(user: UserDTO) {
		sharedPreferences.edit().putString(KEY_USER_REGISTER_CREDENTIAL, Gson().toJson(user)).apply()
	}

	override fun getUser(): UserDTO {
		val json: String = requireNotNull(sharedPreferences.getString(KEY_USER_REGISTER_CREDENTIAL, null))
		return Gson().fromJson(json, UserDTO::class.java)
	}

	companion object {

		const val SHARED_PREFS_USER_REGISTER_CREDENTIAL = "user_register_credential"
		const val KEY_USER_REGISTER_CREDENTIAL = "key_user_register_credential"
	}

}